src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copyState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case `addProperties`:
+        copyState = Object.assign({}, copyState, action.extraData);
+        result.push(copyState);
+        break;
+
+      case `removeProperties`:
+        copyState = { ...copyState };
+
+        for (const removeItem of action.keysToRemove) {
+          delete copyState[removeItem];
+        }
+        result.push(copyState);
+        break;
+
+      case `clear`:
+        copyState = {};
+        result.push(copyState);
```

Comment:
And you can push copyState to your array after switch-case

Reply:
Fixed, thanks for good idea!